File: swing/src/main/java/wannabe/swing/renderer/SquareWithWireSides.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

public class SquareWithWireSides extends SwingRenderer {
  final Sides sides = new Sides();

  @Override public void draw(Graphics g, Projected r) {
    sides.wireSides(g, r);
    g.drawRect(r.left, r.top, r.size, r.size);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/SquareWithWireSides.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class SquareWithWireSides : SwingRenderer() {
  private val sides = Sides()

  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    sides.wireSides(g, p)
    g.drawRect(p.left, p.top, p.size, p.size)
  }
}
